discovery-infra/test_infra/utils.py

```diff
@@ -225,6 +200,22 @@ def are_hosts_in_status(
     return False
 
 
+def are_operators_in_status(operators: List[Dict], operators_count: int, statuses: List[str], fall_on_error_status: bool) -> bool:
+    log.info(
+        "Asked operators to be in one of the statuses from %s and currently operators statuses are %s",
+        statuses,
+        [(operator["name"], operator.get("status"), operator.get("status_info")) for operator in operators],
+    )
+
+    if len([operator for operator in operators if operator.get("status") in statuses]) >= operators_count:
+        return True
+
+    if fall_on_error_status and len([operator for operator in operators if operator.get("status") == consts.OperatorStatus.FAILED]) > 0:
+        raise ValueError("All the operators must be in valid status, but got some in error")
```

Comment:
What do you think about using regular loop instead of list comprehension so we could give information about which operator failed. Now we got `got some in error` and it is vague